zebedee-cms/src/main/java/com/github/onsdigital/zebedee/authorisation/AuthorisationServiceImpl.java

```diff
@@ -58,12 +71,131 @@ public UserIdentity identifyUser(String sessionID) throws UserIdentityException
     }
 
     @Override
-    public DatasetPermissions getUserPermissions(String sessionID, String datasetID, String collectionID) throws DatasetPermissionsException {
-        return null;
+    public DatasetPermissions getUserPermissions(String sessionID, String datasetID, String collectionID)
+            throws DatasetPermissionsException {
+        Session session = getSession(sessionID);
+
+        Collection collection = getCollection(collectionID);
+
+        if (StringUtils.isEmpty(datasetID)) {
+            throw new DatasetPermissionsException("dataset ID required but was empty", SC_BAD_REQUEST);
+        }
+
+        validateCollectionContainsRequestedDataset(collection, datasetID);
+
+        return assignPermissions(session, collection.getDescription(), datasetID);
     }
 
     @Override
     public DatasetPermissions getServicePermissions(String serviceToken) throws DatasetPermissionsException {
         return null;
     }
+
+    DatasetPermissions assignPermissions(Session session, CollectionDescription description, String datasetID)
+            throws DatasetPermissionsException {
+        DatasetPermissions permissions = new DatasetPermissions();
+
+        if (isAdminOrPublisher(session)) {
```

Comment:
will need to explain this one to me. Successful return is true? Error is false?

Reply:
If they are not an admin or publisher it's not a "failure" - they are still a valid user but they are a `viewer` and restricted in what they can see.

From the dataset permissions POV we treat admins and publishers the same - they are allowed to see all prerelease data so we grant them full `CRUD` permissions.

Viewer users have to be in a team which is assigned to a collection to be able to see it's content. So we have to do an addition check to determined if we should grant them any permissions at all. 
If they are in a team assigned to a collection we grant them `R` only, otherwise we grant nothing.

Both the `isAdminOrPublisher` and `isCollectionViewer` methods return `true` or `false` _but_ they can also throw an `IOException` - if this happens something has gone pretty wrong. In such a case we would bail out and return a `500`.